src/screens/LoginScreen.js

```diff
@@ -10,41 +10,106 @@ import { Form, t } from '../components/Form';
 import { imageStyles } from '../styles/images';
 import { layoutStyles, margins } from '../styles/layout';
 import { colors } from '../styles/colors';
+import LoginRequester from '../helpers/requesters/LoginRequester';
 import BackgroundOverlay from '../components/BackgroundOverlay';
 import RaisedContainer from '../components/RaisedContainer';
 import Button from '../components/Button';
 
 export default class LoginScreen extends React.Component {
 
-  static navigationOptions = {
-    tabBarLabel: 'Login',
-    tabBarIcon: ({ tintColor }) => (
-      <Icon name="check" size={20} color="#e91e63" />
-    ),
-  };
-
   constructor(props) {
     super(props);
 
+    this.state = {
+      formValues: this.getInitialFormValues(),
+    }
+
+    this.onFormChange = this.onFormChange.bind(this);
     this.login = this.login.bind(this);
-    this.redirectToSignupScreen = this.redirectToSignupScreen.bind(this);
+    this.navigateToSignupScreen = this.navigateToSignupScreen.bind(this);
+  }
+
+  getInitialFormValues() {
+    return {
+      email: 'kenchen@berkeley.edu',
+      password: 'password',
+    };
+  }
+
+  getFormType() {
+    return t.struct({
+      email: t.String,
+      password: t.String,
+    });
   }
 
+  getFormOptions() {
+    return {
+      fields: {
+        email: {
+          error: 'Cannot be blank',
+          value: 'kenchen@berkeley.edu',
+        },
+        password: {
+          secureTextEntry: true,
+          password: true,
+          error: 'Cannot be blank',
+          value: 'password'
+        },
+      },
+    };
+  }
+
+  onFormChange(values) {
+    this.setState({ formValues: values });
+  }
+
+  /**
+   * Attempts to login the user to the API. If successful then routes the
+   * user to the `App`, otherwise, renders errors.
+   *
+   * @param {function} onSuccess: callback on response when successful
+   * @param {function} onFailure: callback on error object when errored
+   */
   login(event, onSuccess, onFailure) {
     event.preventDefault();
-    const value = this.form.getValue();
-    if (value) {
-      console.log(value);
-      onSuccess && onSuccess(value);
+    const values = this.form.getValue();
+    if (values) {
+      LoginRequester.login(
+        values.email,
+        values.password,
+        onSuccess,
+        onFailure,
+      ).then((response) => {
+        console.log(response);
```

Comment:
in this context, what does `onSuccess` do if it doesnt get called within this clause?